expr/plan.go

```diff
@@ -3,28 +3,101 @@ package expr
 import (
 	"fmt"
 	"io"
+	"strings"
 
 	"github.com/grafana/metrictank/api/models"
 	"github.com/grafana/metrictank/consolidation"
 	"github.com/grafana/metrictank/errors"
 )
 
+type Optimizations struct {
+	PreNormalization bool
+	MDP              bool
+}
+
+func (o Optimizations) ApplyUserPrefs(s string) Optimizations {
+	if s == "" {
+		return o
+	}
+	o.PreNormalization = false
+	o.MDP = false
+	if s == "none" {
+		return o
+	}
+	prefs := strings.Split(s, ",")
```

Comment:
Should we add some error level debugging here in case neither `pn` nor `mdp` are specified (i.e. a typo or similar) to make it easier to debug?

Reply:
i added validation